### PR TITLE
Added with-data flag

### LINKED ISSUE #174: Should `sail rm` remove the directories as well?

### PR DESCRIPTION
**Pull request by roberthmiller**
Resolves #174

### CODE REVIEW

File: rmcmd.go [hunk 1]
```diff
@@ -29,22 +32,23 @@ or all of the containers on a system with the -all flag.`,
...
-
 	c.repoArg = fl.Arg(0)
 
 	if c.repoArg == "" && !c.all {
 		fl.Usage()
 		os.Exit(1)
 	}
 
-	names := c.getRemovalList()
+	// The docker daemon should not be running for the usage command.
```
Comment (nathanpotter) [inline rmcmd.go]: This comment doesn't make any sense

File: rmcmd.go [hunk 2]
```diff
@@ -87,7 +91,13 @@ func removeContainers(names ...string) {
 			flog.Error("failed to remove %s: %v", name, err)
 			continue
 		}
-
+		if c.withData {
+			root := c.gf.config().ProjectRoot
+			if !strings.HasSuffix(root, "/") {
```
Comment (nathanpotter) [inline rmcmd.go]: We should use filepath.Join for these sort of operations

File: rmcmd.go [hunk 3]
```diff
@@ -87,7 +91,13 @@ func removeContainers(names ...string) {
...
+				root = root + "/"
+			}
+			os.RemoveAll(fmt.Sprintf("%s%s", root, c.repoArg))
```
Comment (nathanpotter) [inline rmcmd.go]: this error needs to be checked

File: rmcmd.go [hunk 4]
```diff
@@ -29,22 +32,23 @@ or all of the containers on a system with the -all flag.`,
 }
 
 func (c *rmcmd) RegisterFlags(fl *flag.FlagSet) {
-	fl.BoolVar(&c.all, "all", false, "Remove all sail containers.")
+	fl.BoolVar(&c.all, "all", false, "Remove all Sail containers.")
+	fl.BoolVar(&c.withData, "with-data", false, "Remove all data associated with the Sail instance.")
```
Comment (nathanpotter) [inline rmcmd.go]: I think this comment should specify that it only removes the cloned project directory

File: rmcmd.go (hunk 1 above)
Comment (roberthmiller) [reply rmcmd.go]: I deleted the comment, but my thought process is as follows:

The docker daemon should not be required when it will not be used.

**Pull request merged**